http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -1844,6 +1963,116 @@ private <E extends HttpClientException> E decorate(E exc) {
         return HttpClientExceptionUtils.populateServiceId(exc, informationalServiceId, configuration);
     }
 
+    private @NonNull HttpClientException handleResponseError(io.micronaut.http.HttpRequest<?> finalRequest, Throwable cause) {
+        String message = cause.getMessage();
+        if (message == null) {
+            message = cause.getClass().getSimpleName();
+        }
+        if (log.isTraceEnabled()) {
+            log.trace("HTTP Client exception ({}) occurred for request : {} {}",
+                message, finalRequest.getMethodName(), finalRequest.getUri());
+        }
+
+        HttpClientException result;
+        if (cause instanceof io.micronaut.http.exceptions.ContentLengthExceededException clee) {
+            result = decorate(new ContentLengthExceededException(clee.getMessage()));
+        } else if (cause instanceof io.micronaut.http.exceptions.BufferLengthExceededException blee) {
+            result = decorate(new ContentLengthExceededException(blee.getAdvertisedLength(), blee.getReceivedLength()));
+        } else if (cause instanceof io.netty.handler.timeout.ReadTimeoutException) {
+            result = ReadTimeoutException.TIMEOUT_EXCEPTION;
+        } else if (cause instanceof HttpClientException hce) {
+            result = decorate(hce);
+        } else {
+            result = decorate(new HttpClientException("Error occurred reading HTTP response: " + message, cause));
+        }
+        return result;
+    }
+
+    private static void setRedirectHeaders(@Nullable io.micronaut.http.HttpRequest<?> request, MutableHttpRequest<Object> redirectRequest) {
```

Comment:
can we reduce the number of nested for/if statements by refactoring this method to a make it easier to read

Reply:
possibly, I want to keep it this way in this PR though because it's the same code as before and this PR is already complicated enough